### PR TITLE
Apply projectedName for operation

### PR DESCRIPTION
**Pull request by MaryGao**
Apply the @projectedName for operation which will be used to construct parameter and response interfaces.

```
 @projectedName("json", "testRunOperation")
 op test(): string;
```

typescript parameters.ts

```typescript
import { RequestParameters } from "@azure-rest/core-client";
          
export type TestRunOperationParameters =  RequestParameters;
```

### CODE REVIEW

File: packages/typespec-ts/test/unit/modelsGenerator.spec.ts [hunk 1]
```diff
@@ -1511,6 +1511,24 @@ describe("Input/output model type", () => {
         }`
       });
     });
+
+    it("should generate projected operation name", async () => {
+      const parameters = await emitParameterFromCadl(
+        `
+        @projectedName("javascript", "testRunOperation")
```
Comment (qiaozha) [inline packages/typespec-ts/test/unit/modelsGenerator.spec.ts:1518]: can this still pass ?

File: packages/typespec-ts/test/unit/modelsGenerator.spec.ts [hunk 2]
```diff
@@ -1511,6 +1511,24 @@ describe("Input/output model type", () => {
...
+        op test(): string;
+        `
+      );
+      assert.ok(parameters);
+      assertEqualContent(
+        parameters?.content!,
+        `
+          import { RequestParameters } from "@azure-rest/core-client";
+          
+          export type TestRunOperationParameters =  RequestParameters;
```
Comment (qiaozha) [inline packages/typespec-ts/test/unit/modelsGenerator.spec.ts:1528]: Can you also add a ut for response ?

File: packages/typespec-ts/test/integration/generated/projection/src/clientDefinitions.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ export interface Json {
   ): StreamableMethod<PropertyJson204Response>;
 }
 
-export interface _client {
+export interface _Client {
```
Comment (qiaozha) [inline packages/typespec-ts/test/integration/generated/projection/src/clientDefinitions.ts:28]: why ?

File: packages/typespec-ts/test/unit/modelsGenerator.spec.ts (hunk 1 above)
Comment (MaryGao) [reply packages/typespec-ts/test/unit/modelsGenerator.spec.ts]: updated

File: packages/typespec-ts/test/integration/generated/projection/src/clientDefinitions.ts (hunk 3 above)
Comment (MaryGao) [reply packages/typespec-ts/test/integration/generated/projection/src/clientDefinitions.ts:28]: The interface name is normalized.

File: packages/typespec-ts/test/unit/modelsGenerator.spec.ts (hunk 1 above)
Comment (qiaozha) [reply packages/typespec-ts/test/unit/modelsGenerator.spec.ts]: Maybe keep this as a counter case.

File: packages/typespec-ts/test/unit/modelsGenerator.spec.ts (hunk 2 above)
Comment (MaryGao) [reply packages/typespec-ts/test/unit/modelsGenerator.spec.ts:1532]: done

**Review by qiaozha — APPROVED** [review]

**Pull request merged**